storage/google/cloud/storage/bucket.py

```diff
@@ -586,6 +612,10 @@ def get_blob(self, blob_name, client=None, encryption_key=None, **kwargs):
 
         if self.user_project is not None:
             query_params["userProject"] = self.user_project
+
+        if generation is not None:
+            query_params["generation"] = generation
+
         blob = Blob(
             bucket=self, name=blob_name, encryption_key=encryption_key, **kwargs
         )
```

Comment:
I'd rather create the blob with the expected generation here, and then use its `_query_params` property, e.g.:

```python
        client = self._require_client(client)
        blob = Blob(
            bucket=self, name=blob_name, encryption_key=encryption_key, generation=generation, **kwargs
        )
        query_params = blob._query_params

        try:
            headers = _get_encryption_headers(encryption_key)
            response = client._connection.api_request(
                method="GET",
                path=blob.path,
                query_params=query_params,
                headers=headers,
                _target_object=blob,
            )
```

Now that I look at it more closely, I'm not sure why this method doesn't just call `blob.reload()`.